src/tcache.c

```diff
@@ -581,6 +581,22 @@ tcache_bin_flush_stashed(tsd_t *tsd, tcache_t *tcache, cache_bin_t *cache_bin,
 	assert(head_content == *cache_bin->stack_head);
 }
 
+bool
+tcache_bin_ncached_max_get(tsd_t *tsd, size_t bin_size,
+    cache_bin_sz_t *ncached_max) {
+	if (bin_size > TCACHE_MAXCLASS_LIMIT) {
+		return true;
+	}
+
+	tcache_t *tcache = tsd_tcachep_get(tsd);
+	assert(tcache != NULL);
+	szind_t bin_ind = sz_size2index(bin_size);
+
+	*ncached_max =
+	    cache_bin_ncached_max_get_unsafe(&tcache->bins[bin_ind]);
```

Comment:
If it's past tcache_max, should we return 0? i.e. returning effective ncached_max.

Reply:
This is a good question. My thought is:
1. If we always return the real `ncached_max`, not the effective one, users will have to figure out the real items being cached in a bin by querying `tcache_max` and combine both results themselves;
2. If we return the effective `ncached_max`, users can easily know the actual items to be cached more easily. Yet they have no way to know the real `ncached_max` of a bin when they are disabled, i.e., bin class size larger than `tcache_max`. 

While we design `thread.tcache.ncached_max.write()` to allow users to set a disabled bin's `ncached_max`, I think they should have access to the value they set, i.e, a read after a write should return what they write. So I prefer the first option above, which is the current implementation now.

Reply:
Getting the value they set is a valid point. However this part

> users will have to figure out the real items being cached in a bin by querying tcache_max and combine both

I think is actually a plus, i.e. if they set a high past-tcache_max size class and then read from it immediately, returning 0 is causing the right "confusion", instead of letting people think "everything is set and ready".

One high level design we discussed, was that setting tcache_max and ncached_max in either order should work the same -- which is certainly done here. That's a convenience feature we went far to offer. But beyond that we still want to make the API straighforward and matching the truth as much as possible.

> they have no way to know the real ncached_max of a bin when they are disabled

Honestly I don't think it's very necessary.

Reply:
Agreed. Made the change